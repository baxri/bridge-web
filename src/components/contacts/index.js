import Contacts from './Contacts'

export { default as Contact } from './Contact'
export default Contacts
