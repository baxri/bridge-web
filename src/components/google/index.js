import GoogleNotConnected from './GoogleNotConnected'

export { GoogleNotConnected as default }
