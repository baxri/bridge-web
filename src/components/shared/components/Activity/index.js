import React from 'react'
import styled from 'styled-components'
import { Link } from 'react-router-dom'
import { connect } from 'react-redux'

const StyledActivity = styled.div`
  border-bottom: 1px solid #dfe1e7;
  min-height: 80px;
  color: #000000;
  padding: 12px 12px 18px 12px;
  display: flex;
  font-size: 14px;
`

const Time = styled.div`
  flex: 1;
  opacity: 0.42;
  font-size: 14px;
`

const Activities = styled.div`
  flex: 3;
  a,
  a:hover {
    color: #000000;
    text-decoration: none;
  }
`

const Activity = ({ date, time, activities }) => (
  <StyledActivity>
    <Time>
      <div>{date}</div>
      <div>{time}</div>
    </Time>
    <Activities>
      {activities.map((activity, index) => (
        <Link
          to={`/introductions/${activity.intro_id}`}
          key={`${activity.intro_id}-${activity.time}-${index}`}
        >
          <div
            dangerouslySetInnerHTML={{ __html: activity.text }}
            style={{ marginBottom: 10 }}
          />
        </Link>
      ))}
    </Activities>
  </StyledActivity>
)

const mapStateToProps = (_, { time }) => {
  const d = time.split(/\s(.+)/)

  return {
    time: d[1],
    date: d[0],
  }
}

export default connect(mapStateToProps)(Activity)
