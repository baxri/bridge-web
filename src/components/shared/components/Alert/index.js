import React from 'react'
import styled from 'styled-components'

const Container = styled.div`
  background-color: #f8f8f8;
  padding: 10px;
  font-size: 14px;
  text-align: center;
  margin: 0 -15px;
`

const Alert = React.memo(({ children }) => {
  return <Container>{children}</Container>
})

export default Alert
