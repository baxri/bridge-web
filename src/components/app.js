import React, { Component } from 'react'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'
import { withRouter } from 'react-router-dom'
import { Col } from 'reactstrap'
import { withSnackbar } from 'notistack'
import { compose } from 'redux'
import { keys } from 'lodash'

import FooterTemplate from 'components/layout/footer'
import Nav from 'components/layout/nav'
import Navbar from 'components/layout/Navbar'
import Sidenav from 'components/layout/Sidenav'
import NoAuthNav from 'components/layout/NoAuthNav'
import { isNewUnloggedDesign } from 'utils/newDesign'
import { MixpanelContext } from 'utils/mixpanelContext'
import { LoadingData as Spinner } from 'components/shared'
import 'bootstrap/dist/css/bootstrap.min.css'

import { fetchIntrosCount } from 'intropath-core/actions/counts'
import { RELOAD_TIME } from 'utils/cache'
import { updateAllData } from 'intropath-core/actions/update'
import { syncContacts } from 'intropath-core/actions/contacts'
import { getActivities } from 'intropath-core/actions/activity'
import { showGoogleWarning } from 'intropath-core/actions/user'
import { StyledApp, Wrapper, Content } from './styled'
import GoogleNotConnected from 'components/google'

class App extends Component {
  constructor(props) {
    super(props)

    if (navigator.cookieEnabled) {
      this.noCache = !(
        keys(localStorage).includes('/contacts_compressed') &&
        keys(localStorage).includes('/introductions_compressed')
      )
    }

    this.state = {
      loading: this.noCache,
    }

    this.footerExcludes = ['/introductions/new', '/import-contacts']
  }

  static propsTypes = {
    location: PropTypes.object,
  }

  static contextType = MixpanelContext

  componentWillMount() {
    if (navigator.cookieEnabled) {
      this.fetchAllData()
      this.reloadInterval = setInterval(this.fetchAllData, RELOAD_TIME)
    } else {
      this.props.enqueueSnackbar(
        <h6 className="mb-0">
          Cookies are not enabled on your browser. please enable cookies in your
          browser preferences to continue.
        </h6>,
        {
          variant: 'error',
          persist: true,
        }
      )
    }
  }

  componentWillUnmount() {
    clearInterval(this.reloadInterval)
    window.removeEventListener('focus', this.onPageFocused)
  }

  componentDidMount() {
    window.addEventListener('focus', this.onPageFocused)
  }

  componentDidUpdate(prevProps) {
    if (this.props.location !== prevProps.location) {
      this.onRouteChanged(this.props, prevProps)
    }

    if (this.props.fullFetch && !prevProps.fullFetch) {
      this.props.enqueueSnackbar(
        'Full update. It can take several seconds for the first time.'
      )
    }

    if (this.props.authenticated) {
      const { pathname } = this.props.location
      const syncRoutes = [
        '/',
        '/contacts',
        '/introductions/new',
        '/introductions',
      ]
      if (
        pathname !== prevProps.location.pathname &&
        syncRoutes.includes(pathname)
      ) {
        this.syncContactsAndFetchAllData()
      }

      if (prevProps.user === null && this.props.user !== null) {
        this.fetchAllData()
      }
    }
  }

  onPageFocused = () => {
    this.syncContactsAndFetchAllData()
  }

  syncContactsAndFetchAllData = async () => {
    if (this.props.authenticated) {
      try {
        if (await this.props.syncContacts()) {
          setTimeout(this.fetchAllData, 10000)
          this.props.showGoogleWarning(false)
        }
      } catch (error) {
        this.props.showGoogleWarning(true)
      }
    }
  }

  onRouteChanged(props, prevProps) {
    const { user } = this.props
    const user_id = user ? user.id : ''
    this.context.mixpanel.identify(user_id)
    this.context.mixpanel.track('NAVIGATED_TO', {
      PreviousScreen: prevProps.location.pathname,
      Screen: props.location.pathname,
      UserId: user_id,
    })
  }

  waitForDataToLoad() {
    const { pathname: path } = this.props.location
    return path === '/' || path === '/introductions' || path === '/contacts'
  }

  noAuthRequired() {
    const { pathname: path } = this.props.location
    return (
      path.includes('accept') ||
      path.includes('confirm') ||
      path.includes('new-by-share') ||
      path.includes('rate') ||
      path.includes('rating-message') ||
      path.includes('recover') ||
      path.includes('cancel') ||
      path.includes('reject') ||
      path.includes('cancel_by_owner') ||
      path.includes('reset-password')
    )
  }

  canceledByOwner() {
    const { pathname: path } = this.props.location
    return path.includes('cancel_by_owner')
  }

  shouldFetchAllData() {
    return this.noAuthRequired()
  }

  fetchAllData = () => {
    if (
      this.props.authenticated &&
      (this.canceledByOwner() || !this.shouldFetchAllData())
    ) {
      if (!this.noCache) {
        // when fetchAllData is not the first time
        this.props.enqueueSnackbar('Loading...')
        this.props.fetchIntrosCount({
          query: this.props.query,
        })
        this.props.getActivities()
      }

      this.props
        .updateAllData(false, true)
        .then(() => this.setState({ loading: false }))
        .catch(() => this.setState({ loading: false }))
    }
  }

  renderNav() {
    const path = this.props.location.pathname
    const isNoAuthPage = ''
    if (isNewUnloggedDesign(path)) return

    if (this.noAuthRequired()) {
      const { pathname: path } = this.props.location
      const recover_path = path.includes('recover')

      return <NoAuthNav recover_path={recover_path} />
    } else {
      return isNoAuthPage ? <Nav /> : <Navbar />
    }
  }

  renderContent() {
    const {
      location: { pathname },
      authenticated,
    } = this.props
    const isFull =
      this.footerExcludes.includes(pathname) || this.noAuthRequired()

    if (isNewUnloggedDesign(pathname)) {
      return this.props.children
    } else if (
      authenticated &&
      this.state.loading &&
      this.waitForDataToLoad()
    ) {
      return <Spinner loading={true} />
    } else {
      return (
        <Wrapper className="row">
          {!isFull && (
            <Col lg={2}>
              {!this.noAuthRequired() && <Sidenav {...this.props} />}
            </Col>
          )}

          <Content lg={isFull ? 12 : 10} pathname={pathname}>
            <GoogleNotConnected />
            {this.props.children}
          </Content>
        </Wrapper>
      )
    }
  }

  render() {
    const { pathname } = this.props.location
    const isNewUnloggedDesign =
      pathname.includes('login') ||
      pathname.includes('forgot-password') ||
      pathname.includes('register')
    let appClass = 'app-container'
    const tankYouPage =
      pathname.includes('confirmation-success') ||
      pathname.includes('accept') ||
      pathname.includes('cancel') ||
      pathname.includes('reject') ||
      pathname.includes('rating-message')
    if (tankYouPage) {
      appClass = 'app-container thank-you-page'
    } else if (isNewUnloggedDesign) {
      appClass = 'container app-container'
    }

    return (
      <StyledApp>
        {this.renderNav()}

        <div className={appClass}>{this.renderContent()}</div>

        {!this.footerExcludes.includes(pathname) &&
          !pathname.includes('/confirm') && <FooterTemplate />}
      </StyledApp>
    )
  }
}

const mapStateToProps = ({
  auth,
  filter,
  introduction: { fullFetch },
  contacts: { fullFetch: contactsFullFetch },
}) => ({
  authenticated: auth.authenticated,
  user: auth.user,
  query: filter.introduction.query,
  fullFetch: fullFetch || contactsFullFetch,
})

const enhance = compose(
  withSnackbar,
  withRouter,
  connect(
    mapStateToProps,
    {
      updateAllData,
      syncContacts,
      fetchIntrosCount,
      getActivities,
      showGoogleWarning,
    }
  )
)

export default enhance(App)
