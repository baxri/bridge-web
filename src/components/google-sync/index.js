import React from 'react'
import { withSnackbar } from 'notistack'
import { connect } from 'react-redux'
import queryString from 'query-string'
import {
  GoogleSyncWrapper,
  Body,
  HandShakeImage,
  HandShakeLogoWrapper,
  Title,
  Text,
  PrivacyText,
  GoogleButton,
  ButtonText,
  GoogleIcon,
  NotificationContainer,
  NotificationText,
  NotificationSubText,
} from './styled'
import Header from './Header'
import ContactsImport from './ContactsImportForm'

import googleOAuth from 'utils/googleOAuth'
import { createContacts } from 'intropath-core/actions/contacts'
import {
  addToken,
  fetchAccessToken,
  updateUser,
} from 'intropath-core/actions/user'
import importContacts from 'utils/importContacts'

class GoogleSync extends React.PureComponent {
  state = {
    isGoogleSignIn: false,
    totalContacts: 0,
    importedContacts: 0,
    syncing: false,
  }

  componentDidMount() {
    this.query = queryString.parse(this.props.location.search)
    if (this.query.importing) this.onGoogleSignIn()
  }

  onClose = () => {
    if (this.query.redirectOnClose) {
      this.props.history.push(this.query.redirectOnClose)
    } else {
      this.props.history.goBack()
    }
  }

  redirectAfterFinish = () => {
    this.query = queryString.parse(this.props.location.search)
    if (this.query.redirectAfter) {
      this.props.history.push(this.query.redirectAfter)
    } else {
      this.onClose()
    }
  }

  onGoogleSignIn = () => {
    googleOAuth()
      .then(({ token, profile_pic_url }) => {
        const { user, addToken, updateUser } = this.props
        addToken(token)
        if (profile_pic_url) {
          updateUser(user.id, {
            profile_pic_url,
            pic_type: 'token',
            token_id: token.id,
          })
        }

        return token
      })
      .then(token => this.importGoogleContacts(token))
      .catch(err => {
        if (err && err.error && err.error === 'popup_blocked_by_browser') {
          window.alert(
            'Popup is required to connect google account. Please, allow popup in browser and try again.'
          )
        } else {
          window.Raven.captureException(JSON.stringify(err))
        }

        console.log(err)
        this.props.enqueueSnackbar(
          'Error: Contacts did not sync. ' +
            (err && err.error ? 'Reason: ' + err.error : ''),
          {
            variant: 'error',
            autoHideDuration: 10000,
          }
        )
      })
  }

  importGoogleContacts = token => {
    if (!token) return

    let contacts = []

    this.setState({ isGoogleSignIn: true })
    fetchAccessToken(token.id)
      .then(({ data }) => {
        return importContacts(
          data.access_token,
          totalContacts => this.setState({ totalContacts }),
          newContacts => {
            contacts = contacts.concat(newContacts)
            const importedContacts =
              contacts.length > this.state.totalContacts
                ? this.state.totalContacts
                : contacts.length
            this.setState({ importedContacts })
          }
        )
      })
      .then(() => {
        // TODO Because contacts might have 0, 1 or more emails then import contact
        // will not match with total contact count so this is a hacky workaround for now
        this.setState({
          syncing: true,
          importedContacts: this.state.totalContacts,
        })

        if (contacts.length > 0) {
          this.props
            .createContacts({ contacts, token_id: token.id })
            .then(() => {
              this.redirectAfterFinish()
            })
        } else {
          this.redirectAfterFinish()
        }
      })
      .catch(e => {
        this.setState({ isGoogleSignIn: false })
        window.Raven.captureException(JSON.stringify(e))
      })
  }

  render() {
    const { tokenIsIvalid } = queryString.parse(this.props.location.search)
    const { totalContacts, importedContacts, syncing } = this.state
    return (
      <GoogleSyncWrapper>
        <Header onClose={this.onClose} title="Sync Contacts" />
        {!this.state.isGoogleSignIn && tokenIsIvalid === '1' && (
          <NotificationContainer>
            <NotificationText>
              Your connected google account needs your attention.{' '}
              <NotificationSubText>
                Please sync contacts again.
              </NotificationSubText>
            </NotificationText>
          </NotificationContainer>
        )}
        <Body>
          {this.state.isGoogleSignIn ? (
            <ContactsImport
              total={totalContacts}
              importedContacts={importedContacts}
              syncing={syncing}
            />
          ) : (
            <div>
              <HandShakeLogoWrapper>
                <HandShakeImage src="/img/handshake.svg" />
              </HandShakeLogoWrapper>
              <Title>Bridge makes introductions easy.</Title>
              <Text>
                In order to help you make introductions that are personal and
                affective, Bridge needs your permission to send emails from your
                account.
              </Text>
              <GoogleButton onClick={this.onGoogleSignIn}>
                <ButtonText>Sync Your Contacts</ButtonText>
                <GoogleIcon src="/img/icons/icon-google.svg" />
              </GoogleButton>
              <PrivacyText>
                At no time does Bridge have access to your emails. Think of
                Bridge as a personal assistant to help you send and manage
                intros on your behalf. You can revoke access at any time and
                delete your account if Bridge is no longer useful for you.
              </PrivacyText>
            </div>
          )}
        </Body>
      </GoogleSyncWrapper>
    )
  }
}

function mapStateToProps({ auth }) {
  return {
    user: auth.user,
  }
}

export default connect(
  mapStateToProps,
  { createContacts, addToken, updateUser }
)(withSnackbar(GoogleSync))
