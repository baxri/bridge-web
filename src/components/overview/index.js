import React from 'react'
import { Row, Col } from 'reactstrap'
import { connect } from 'react-redux'
import styled from 'styled-components'
import media from 'utils/styledMediaQueryFix'

import Cards from './Cards'
import RecentIntros from './RecentIntros'
import LatestActivity from './LatestActivity'
import { Heading, Button } from 'components/shared'
import ConfirmationAlert from 'components/auth/ConfirmationAlert'
import { checkUserTokens } from 'utils/checkGoogleAccount'
import { getItem } from 'utils/storage'
import { fetchUser } from 'intropath-core/actions/user'

const Background = styled(Col)`
  ${media.lessThan('large')`
    margin-bottom: 10px;
  `}
`

class OverviewPage extends React.Component {
  componentWillMount() {
    const { deleting, history, fetchUser } = this.props
    if (deleting) history.push('/recover')

    const user = getItem('user')
    if (user) fetchUser(user.id)
  }

  onStartIntroClick = () => {
    const { user, history } = this.props
    checkUserTokens(user.tokens, history, '/introductions/new')
  }

  render() {
    const { introductions, contacts, loaded } = this.props

    return (
      <div>
        <ConfirmationAlert />
        <Heading.HeaderAction text="Overview">
          <Button full onClick={this.onStartIntroClick}>
            Make an intro
          </Button>
        </Heading.HeaderAction>

        {!loaded && <span>Loading...</span>}
        {loaded && (
          <div>
            <Cards
              numberOfContacts={contacts.length}
              {...{ introductions }}
              onClick={this.onStartIntroClick}
            />

            <Row style={{ marginTop: 5 }}>
              <Background lg={7}>
                <RecentIntros {...{ introductions }} />
              </Background>
              <Col lg={5}>{loaded && <LatestActivity />}</Col>
            </Row>
          </div>
        )}
      </div>
    )
  }
}

const mapStateToProps = state => {
  return {
    user: state.auth.user,
    authenticated: state.auth.authenticated,
    deleting: state.auth.deleting,
    introductions: state.introduction.list,
    contacts: state.contacts.list,
    loaded: state.introduction.loaded && state.contacts.loaded,
  }
}

export default connect(
  mapStateToProps,
  { fetchUser }
)(OverviewPage)
