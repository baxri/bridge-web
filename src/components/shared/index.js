import { ArrowAnchor } from './components/Button'
// import Button from './components/Button'
import Button from './components/Button-v2'
import CardOverview from './components/Card/CardOverview'
import Intro from './components/Intro'
import * as Heading from './components/Heading'
import Avatar from './components/Avatar'
import Activity from './components/Activity'
import * as Icons from './components/Icons'
import TimeAgo from './components/Time/TimeAgo'
import StyledInput from './components/Input/StyledInput'
import AuthBodyPanel from './components/AuthPages/BodyPanel'
import Header from './components/AuthPages/Header'
import StyledCol from './components/StyledCol'
import TabItem from './components/Tabs/TabItem'
import SearchInput from './components/Input/SearchInput'
import SelectedContact from './components/Contact/SelectedContact'
import * as EmptyState from './components/EmptyState'
import Field from './components/Input/Field'
import Paginate from './components/Paginate/Paginate'
import Contact from './components/Contact/Contact'
import WithDropdown from './components/Dropdown/WithDropdown'
import IntroIconState from './components/Intro/IconState'
import IntroContext from './components/Intro/IntroContext'
import SelectedContacts from './components/Contact/SelectedContacts'
import Flash from './components/Flash'
import AutosuggestTagsInputField from './components/Autosuggest/AutosuggestTagsInputField'
import FormGroupField from './components/Form/FormGroupField'
import Textarea from './components/Form/Textarea'
import Spinner, { LoadingData } from './components/Spinner'
import IntroInput from './components/IntroInput/IntroInput'
import Thankyou from './components/Thankyou/Thankyou'
import Alert from './components/Alert'
import Confirmation from './components/Confirmation'
import { YButton, NButton } from './components/Confirmation/styled'

export {
  Activity,
  Alert,
  ArrowAnchor,
  AuthBodyPanel,
  AutosuggestTagsInputField,
  Avatar,
  Button,
  CardOverview,
  Confirmation,
  Contact,
  EmptyState,
  Field,
  Flash,
  FormGroupField,
  Header,
  Heading,
  Icons,
  Intro,
  IntroContext,
  IntroIconState,
  IntroInput,
  LoadingData,
  NButton,
  Paginate,
  SearchInput,
  SelectedContact,
  SelectedContacts,
  Spinner,
  StyledCol,
  StyledInput,
  TabItem,
  Textarea,
  Thankyou,
  TimeAgo,
  WithDropdown,
  YButton,
}
