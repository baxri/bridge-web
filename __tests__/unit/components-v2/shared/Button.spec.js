import React from 'react'
import { shallow } from 'enzyme'

import { Button } from 'components/shared'

describe('<Button />', () => {
  it('render button', () => {
    const btn = shallow(<Button>Button</Button>)
    expect(btn.props().children).toBe('Button')
  })

  it('have borderless', () => {
    const btn = shallow(<Button borderless>Button</Button>)
    expect(btn).toMatchSnapshot()
  })
})
