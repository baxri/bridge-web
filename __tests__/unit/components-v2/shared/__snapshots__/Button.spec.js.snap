// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> have borderless 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    borderless={true}
  >
    Button
  </Button>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "borderless": true,
      "children": "Button",
    },
    "ref": null,
    "rendered": "Button",
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bxivhb",
        "isStatic": false,
        "rules": Array [
          "
  background: ",
          [Function],
          ";
  color: ",
          [Function],
          ";
  text-align: center;
  border: 1px solid ",
          [Function],
          ";
  border-radius: ",
          [Function],
          ";
  min-width: ",
          [Function],
          ";
  height: ",
          [Function],
          ";
  font-size: ",
          [Function],
          ";
  opacity: ",
          [Function],
          ";
  padding: ",
          [Function],
          ";
  width: ",
          [Function],
          ";
  font-style: normal;

  &:active {
    background: ",
          [Function],
          ";
  }
",
        ],
      },
      "displayName": "styled.button",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bxivhb",
      "target": "button",
      "toString": [Function],
      "usesTheme": false,
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "borderless": true,
        "children": "Button",
      },
      "ref": null,
      "rendered": "Button",
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bxivhb",
          "isStatic": false,
          "rules": Array [
            "
  background: ",
            [Function],
            ";
  color: ",
            [Function],
            ";
  text-align: center;
  border: 1px solid ",
            [Function],
            ";
  border-radius: ",
            [Function],
            ";
  min-width: ",
            [Function],
            ";
  height: ",
            [Function],
            ";
  font-size: ",
            [Function],
            ";
  opacity: ",
            [Function],
            ";
  padding: ",
            [Function],
            ";
  width: ",
            [Function],
            ";
  font-style: normal;

  &:active {
    background: ",
            [Function],
            ";
  }
",
          ],
        },
        "displayName": "styled.button",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bxivhb",
        "target": "button",
        "toString": [Function],
        "usesTheme": false,
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
